modules/urlsigning-service-impl/src/main/java/org/opencastproject/security/urlsigning/provider/impl/AbstractUrlSigningProvider.java

```diff
@@ -208,9 +224,20 @@ public void updated(Dictionary<String, ?> properties) throws ConfigurationExcept
     }
 
     this.urls = urls;
+    this.exclusionPattern = exclusionPattern;
+  }
+
+  private boolean isExcluded(String url) {
+    boolean isExcluded = false;
+    Pattern exclusionPattern = this.exclusionPattern;
```

Comment:
This assignment is unnecessary, right? :thinking:

Reply:
No, the assignment is required to make the code thread-safe:

```
    if (exclusionPattern != null) {
      // When the configuration changes, exclusionPattern might by null here
      Matcher matcher = exclusionPattern.matcher(url);
```

Reply:
Why, what would modify `this.exclusionPattern` here? I think nothing could. The docs specifically state:

Instances of this class (`Pattern`) are immutable and are safe for use by multiple concurrent threads. Instances of the Matcher class are not safe for such use.

Reply:
Btw even if the pattern was not immutable, copying a reference to it would not make the code thread safe.

Am I missing something here? :thinking:

Reply:
Ooh yes I am, I take everything back, see my comment about the weather above. :sweat_smile:

Reply:
> Btw even if the pattern was not immutable, copying a reference to it would not make the code thread safe.

As far as I know, assigning references is an atomic operation in Java. Therefore, the code `return exclusionPattern != null && exclusionPattern.matcher(url)` is thread-safe for the local variable `exclusionPattern` but not for the variable `this.exclusionPattern` because that variable is set whenever the configuration file changes.

Reply:
Yes, you are right. I was thinking you were trying to protect `this.exclusionPattern` from having its internal state mutated by another thread during this operation, which just creating another reference would not help you with. But that doesn't matter anyway, since `Pattern` is immutable. And your code correctly protects us from the reference itself changing under us between checking it and calling methods on it, I see that now.

Reply:
Admittedly, the case is highly unlikely and wouldn't cause much damage... but still, there is a difference between software that is almost correct and software that is 100% correct ;-)